packages/gatsby/src/redux/reducers/jobsv2.ts

```diff
@@ -16,7 +26,9 @@ module.exports = (
 
     case `END_JOB_V2`: {
       const { jobContentDigest, result } = action.payload
-      const { job } = state.incomplete.get(jobContentDigest)
+      const { job = jobDefaultValue } = state.incomplete.get(
```

Comment:
Likewise here. If the job is undefined then there's nothing to remove. You don't want to add a non-existent job to the "complete" queue.